dist/backdraft.js

```diff
@@ -1303,6 +1303,19 @@ _.extend(Plugin.factory, {
       }
     },
 
+    // takes a hash of { columnTitle: columnState, ... }
+    setColumnVisibilities: function(columns) {
+      _.each(columns, function(state, title) {
+        if (!state && this._columnManager.columnConfigForTitle(title).required) {
+          throw new Error("can not disable visibility when column is required");
+        }
+      }, this);
+      this._columnManager.visibility.set(columns);
```

Comment:
is this line critical to do in bulk like you are doing here?  Because otherwise, this whole method could become:

```
setColumnVisibilities: function(columns) {
  _.each(columns, function(state, title) {
    this.columnVisibility(title, state);
  }
}
```

That's way DRYer too..